zcash_client_backend/src/keys.rs

```diff
@@ -28,18 +45,223 @@ pub fn spending_key(seed: &[u8], coin_type: u32, account: u32) -> ExtendedSpendi
         &[
             ChildIndex::Hardened(32),
             ChildIndex::Hardened(coin_type),
-            ChildIndex::Hardened(account),
+            ChildIndex::Hardened(account.0),
         ],
     )
 }
 
+#[cfg(feature = "transparent-inputs")]
+pub fn derive_transparent_address_from_secret_key(
+    secret_key: &secp256k1::key::SecretKey,
+) -> TransparentAddress {
+    let secp = Secp256k1::new();
+    let pk = PublicKey::from_secret_key(&secp, secret_key);
+    derive_transparent_address_from_public_key(&pk)
+}
+
+#[cfg(feature = "transparent-inputs")]
+pub fn derive_transparent_address_from_public_key(
+    public_key: &secp256k1::key::PublicKey,
+) -> TransparentAddress {
+    let mut hash160 = ripemd160::Ripemd160::new();
+    hash160.update(Sha256::digest(&public_key.serialize()));
+    TransparentAddress::PublicKey(*hash160.finalize().as_ref())
+}
+
+#[cfg(feature = "transparent-inputs")]
+pub fn derive_secret_key_from_seed<P: consensus::Parameters>(
+    params: &P,
+    seed: &[u8],
+    account: AccountId,
+    index: u32,
+) -> Result<SecretKey, hdwallet::error::Error> {
+    let private_key =
+        derive_extended_private_key_from_seed(params, seed, account, index)?.private_key;
+    Ok(private_key)
+}
+
+#[cfg(feature = "transparent-inputs")]
+pub fn derive_public_key_from_seed<P: consensus::Parameters>(
```

Comment:
Document this.

Reply:
fixed in https://github.com/zcash/librustzcash/pull/341/commits/ffc4d0cefbf53b846a5e7ce3dd5f88c45b107861